scim-core/src/main/java/org/apache/directory/scim/core/repository/PatchHandlerImpl.java

```diff
@@ -104,28 +92,19 @@ private <T extends ScimResource> T apply(final T source, final PatchOperation pa
     if (valuePathExpression.getAttributeExpression() != null && attributeObject instanceof Collection<?>) {
       // apply expression filter
       Collection<Object> items = (Collection<Object>) attributeObject;
-      Collection<Object> updatedCollection = items.stream().map(item -> {
-        KeyedResource keyedResource = (KeyedResource) item;
```

Comment:
@erant10 I think this was a `KeyedResource` because it was a common parent, but nothing is requiring these objects to be that type, so I changed that method to return a `Predicate<Object>` instead.